sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTranslator.java

```diff
@@ -0,0 +1,378 @@
+                  ServiceAccountJwtAccessCredentials.newBuilder()
+                      .setClientEmail(serviceAccount)
+                      .setPrivateKey(privateKey)
+                      .build();
+              builder.setCredentialFactory(FixedCredentialFactory.create(credentials));
+            } catch (GeneralSecurityException exception) {
+              throw new RuntimeException("exception while retrieving credentials", exception);
+            }
+            break;
+          case SuppliedCredentials:
+            Credentials credentials =
+                ((CredentialOptions.UserSuppliedCredentialOptions) credOptions).getCredential();
+            builder.setCredentialFactory(FixedCredentialFactory.create(credentials));
+            break;
+          case SuppliedJson:
+            CredentialOptions.JsonCredentialsOptions jsonCredentialsOptions =
+                (CredentialOptions.JsonCredentialsOptions) credOptions;
+            builder.setCredentialFactory(
+                FixedCredentialFactory.create(
+                    GoogleCredentials.fromStream(jsonCredentialsOptions.getInputStream())));
+            break;
+          case None:
+            // pipelineOptions is ignored
+            PipelineOptions pipelineOptions = PipelineOptionsFactory.create();
+            builder.setCredentialFactory(NoopCredentialFactory.fromOptions(pipelineOptions));
+            break;
+        }
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to translate BigtableOptions to BigtableConfig", e);
+      }
+    }
+
+    return builder.build();
+  }
+
+  /** Translate BigtableOptions to BigtableReadOptions. */
+  static BigtableReadOptions translateToBigtableReadOptions(
+      BigtableReadOptions readOptions, BigtableOptions options) {
+    BigtableReadOptions.Builder builder = readOptions.toBuilder();
+    if (options.getCallOptionsConfig().getReadStreamRpcAttemptTimeoutMs().isPresent()) {
+      builder.setAttemptTimeout(
+          org.joda.time.Duration.millis(
+              options.getCallOptionsConfig().getReadStreamRpcAttemptTimeoutMs().get()));
+    }
+    builder.setOperationTimeout(
+        org.joda.time.Duration.millis(options.getCallOptionsConfig().getReadStreamRpcTimeoutMs()));
+    return builder.build();
+  }
+
+  /** Translate BigtableOptions to BigtableWriteOptions. */
+  static BigtableWriteOptions translateToBigtableWriteOptions(
+      BigtableWriteOptions writeOptions, BigtableOptions options) {
+
+    BigtableWriteOptions.Builder builder = writeOptions.toBuilder();
+    // configure timeouts
+    if (options.getCallOptionsConfig().getMutateRpcAttemptTimeoutMs().isPresent()) {
+      builder.setAttemptTimeout(
+          org.joda.time.Duration.millis(
+              options.getCallOptionsConfig().getMutateRpcAttemptTimeoutMs().get()));
+    }
+    builder.setOperationTimeout(
+        org.joda.time.Duration.millis(options.getCallOptionsConfig().getMutateRpcTimeoutMs()));
```

Comment:
The imports imported threeten.bp because our library and beam use different time libraries. Should I swap the imports to use joda.time? does it matter?

Reply:
ahh I see, sorry missed that